### ISSUE TITLE
Meta-Transactions support

### ISSUE THREAD

**Issue by anthony-near** [near/near-cli-rs#141]
CLI support for meta transactions as specified in [NEP-366](https://github.com/near/NEPs/pull/366)

Specific implementation TBD.

**Comment by frol** [comment]
@anthony-near I am happy to discuss how exactly we will integrate it into CLI. There are definitely a ton of opportunities here!

**Comment by frol** [comment]
Just to give some context about both options:

The first option (1):

I expect the relayer to mimic the NEAR JSON RPC interface and just intercept `broadcast_tx*` functions while routing all the other RPC calls as-is. This way it is completely transparent to the CLI users and the gas fees are paid by the relayer. I think it could be a viable solution for "Pagoda+" if one would exist like Aurora+.

The second option (2):

Instead of requiring relayer node to be a transparent proxy we can define a completely independent API with just a couple of methods (blocking and non-blocking calls just like `broadcast_tx*` has) to relay a transaction, and then from  CLI point of view, at the last step where we currently ask user for confirmation to send a transaction we can extend it with an option to send via relayer. Here is the current flow:

<img width="1317" alt="image" src="https://user-images.githubusercontent.com/304265/206248324-3ee8d4d7-a8e8-4c3d-8180-7427fe812def.png">

So in addition to `send` and `print` we can have `send-via-relay` (I believe we should consider how to frame it properly, as the current naming proposal is implementation-driven)

The second option is more explicit and I would probably go this route.

**Comment by firatNEAR** [comment]
I also think that the second option would be a better option for now. User could be explicit about sending the transaction via the relayer.

**Comment by frol** [comment]
@anthony-near @firatNEAR Meta-transaction support is implemented in #189! After some brainstorming with @FroVolod we decided to go with a fully-transparent mode, so users won't be bothered to make a choice whether to send it on their own or through a delegation relay. Given that near-cli-rs has support for several network-connection configs, "free transactions"-providers can offer users to set up a new connection like [this](https://github.com/near/near-cli-rs/blob/master/docs/GUIDE.en.md#add-connection---add-a-network-connection):

```
near config \
    add-connection \
        --network-name testnet \
        --connection-name pagoda-testnet \
        --rpc-url https://near-testnet.api.pagoda.co/rpc/v1/ \
        --wallet-url https://wallet.testnet.near.org/ \
        --explorer-transaction-url https://explorer.testnet.near.org/transactions/ \
        --rpc-api-key 'c0a25b3c-39c2-4f62-a621-50e208b88e64' \
        --linkdrop-account-id testnet \
        --faucet-url https://helper.nearprotocol.com/account \
        --meta-transaction-relayer-url https://near-testnet.api.pagoda.co/relay
```

(Note that `--meta-transaction-relayer-url` is not a real service, I made this URL up for the demo purposes)

When the network-connection is configured and has `meta-transaction-relayer-url`, all the transactions submitted through this network-connection will be submitted through the relayer:

<img width="1510" alt="image" src="https://user-images.githubusercontent.com/304265/235742415-222298ee-ec0a-4e69-94ba-434782f09bf8.png">

Notice that once "meta-testnet" was selected, the hint messages were turned to refer to signing and sending "delegate action" instead of a "transaction".

There are no relayer services I am aware of that support this relayer protocol (`{"signed_delegate_action": "base64-encoded-string-of-borsh-serialized-signed-delegate-action"}`, so it is not a surprise that it failed with "Request failed with status code: 422 Unprocessable Entity". Here is how you can, however manually relay the delegated action with near-cli-rs:

1. Prepare the delegate action from `frol14.testnet` to `frol13.testnet`:
<img width="1508" alt="image" src="https://user-images.githubusercontent.com/304265/235744090-1df2f6a4-58b0-4334-b4c1-af6f8ee6e3a5.png">

2. Relay the signed delegate action (base64-encoded borsh-serialized delegate action) by `qq.frol10.testnet`:
<img width="1508" alt="image" src="https://user-images.githubusercontent.com/304265/235744384-74d0ba6f-92ed-4af2-bb78-436251050cfc.png">
